### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by FeLL1kS**

### CODE REVIEW

File: src/app/article/article/article.component.html [hunk 1]
```diff
@@ -15,6 +15,15 @@ <h1>{{ article.title }}</h1>
           <span class="counter"></span>
         </button>
         &nbsp;&nbsp;
+        <button
+          *ngIf="isCreatedByCurrentUser$ | async | async"
```
Comment (FeLL1kS) [inline src/app/article/article/article.component.html]: Is there any way to avoid this?

File: src/app/article/article/article.component.ts [hunk 2]
```diff
@@ -11,18 +12,39 @@ import { ArticleService } from 'src/app/article/article.service';
 })
 export class ArticleComponent implements OnInit {
   article$ = this.articleService.article$;
+  comments$ = this.articleService.comments$;
+  user$ = this.articleService.user$;
   favoritedArticle$ = this.article$.pipe(map((article) => article?.favorited));
 
+  isCreatedByCurrentUser$ = this.user$.pipe(
```
Comment (FeLL1kS) [inline src/app/article/article/article.component.ts:19]: Here I am trying to compare the user from store and the author of the article

File: src/app/article/article/article.component.html (hunk 1 above)
Comment (FeLL1kS) [reply src/app/article/article/article.component.html]: I found a solution, I need to use concatMap

File: src/app/article/article/article.component.ts (hunk 2 above)
Comment (FeLL1kS) [reply src/app/article/article/article.component.ts:19]: I found a solution, I need to use concatMap

File: src/app/article/article/article.component.ts [hunk 3]
```diff
@@ -11,18 +12,39 @@ import { ArticleService } from 'src/app/article/article.service';
...
+    concatMap((user) =>
```
Comment (anton62k) [inline src/app/article/article/article.component.ts:20]: Just for information: there are concatMap and mergeMap https://www.learnrxjs.io/learn-rxjs/operators/transformation/concatmap but I haven't checked which one is better in your case.

**Review by anton62k — APPROVED** [review]

File: src/app/article/article/article.component.ts (hunk 3 above)
Comment (FeLL1kS) [reply src/app/article/article/article.component.ts:20]: Thanks

**Pull request merged**